### PR TITLE
feat(downloads): extend 'Clear done' to ready-for-merge groups

### LINKED ISSUE #139: Clear non-active downloads: extend 'Clear finished' to cover ready-for-merge and deleted-file groups
**Priority:** Medium | **Effort:** Small

## Motivation

The Downloads tab accumulates entries that users cannot dismiss. Two stuck states are common:

1. **Ready-for-merge with deleted files** — a user downloads an episode (auto-merge off), then later deletes the `.mp4` from AnimeDetailView or from disk. The group stays in \"ready for merge\" status forever because `clearCompleted()` explicitly skips unmerged groups, and there is no per-row remove button.
2. **Ready-for-merge intentionally never merged** — a user downloads several episodes but decides not to merge them (e.g. prefers raw `.mp4`). The groups pile up with no way to clear the queue entry.

Currently the \"Clear finished\" button only appears for merged/failed groups and only removes those. There is no escape hatch for completed-but-unmerged downloads.

## Approach & Architecture

Extend the existing `download:clear-completed` IPC path to treat *all terminal, non-progressing states* as clearable:

- **Cancelled** — already cleared today
- **Failed** — already cleared today
- **Merged** (`mergeStatus === 'completed'`) — already cleared today
- **Merge-failed** (`mergeStatus === 'failed'`) — already cleared today
- **Ready-for-merge** (`all items completed`, no merge entry in `mergeStatuses`) — **new**

Files on disk are left untouched when clearing a ready-for-merge group; only the in-memory queue entry (and its persisted `queue.json` record) is removed.

No new IPC channel is needed.

## UI / UX Considerations

- The \"Clear finished\" button in `DownloadsView` currently appears when `hasFinished` is true (merged or failed groups present). Extend the condition to also trigger on `ready-for-merge` groups.
- Rename the button label from **"Clear finished"** to **"Clear done"** so it accurately reflects the broader set of states it now covers.
- The button remains a bulk action — it clears *all* clearable groups at once. No per-row change.

## Implementation Plan

1. **Main Process (`src/main/download-manager.ts`):**
   - [ ] In `clearCompleted()`, change the merge-status guard from `merge?.status === 'completed' || merge?.status === 'failed'` to `!merge || merge.status === 'completed' || merge.status === 'failed'`. This adds the `!merge` (undefined = never merged = ready-for-merge) case to the removable set.

2. **Preload & IPC (`src/preload/`):**
   - *No changes required* — `download:clear-completed` already exists and is already wired.

3. **Renderer (`src/renderer/src/components/DownloadsView.vue`):**
   - [ ] Update `hasFinished` computed to also return `true` when any group has status `'ready-for-merge'`.
   - [ ] Rename the button label from `"Clear finished"` to `"Clear done"`.

## Files to Touch

- **Modify:** `src/main/download-manager.ts`
- **Modify:** `src/renderer/src/components/DownloadsView.vue`
- **Documentation:** `DESIGN.md` — update the `download:clear-completed` channel description (remove the \"skips unmerged\" qualifier)

## Testing Strategy

- Download an episode with auto-merge **off**. Verify the group shows \"ready for merge\" and the \"Clear done\" button appears.
- Click \"Clear done\" — confirm the group disappears from the list and the `.mp4` file is **still present** on disk.
- Download an episode with auto-merge **off**, then delete the `.mp4` via AnimeDetailView. Confirm the group is still visible, click \"Clear done\", and confirm it disappears.
- Download and fully merge an episode. Confirm \"Clear done\" still removes it (regression check for the existing merged-clear path).
- Download an episode, let it fail. Confirm \"Clear done\" still removes it (regression check for failed-clear path).
- Verify no items currently downloading or queued are accidentally cleared.

## Risks & Edge Cases

- **Merging in progress (`mergeStatus === 'merging'`)**: not affected — `!merge` is false and `merge.status` is neither `'completed'` nor `'failed'`, so currently-merging groups remain in the queue. No regression here.
- **Partial downloads (paused/queued)**: not affected — only items with `item.status === 'completed'` enter the merge-status branch. Paused/queued items remain.
- **queue.json persistence**: `clearCompleted()` already calls `schedulePersist()`, so the cleared groups will not re-appear after restart.

## Out of Scope

- Per-row individual dismiss/remove buttons (explicitly deferred — bulk clear is sufficient for now).
- Deleting on-disk files when clearing a ready-for-merge group (user prefers files to remain).
- A separate \"Clear unmerged\" button distinct from \"Clear done\".

### PR DESCRIPTION
**Pull request by akvasha**
## Summary

Closes #139. The Downloads tab "Clear finished" button is renamed **"Clear done"** and now also dismisses completed downloads that were never merged (auto-merge off, or the `.mp4` deleted from disk leaving the group stuck on "ready for merge" forever). Files on disk are untouched — only the in-memory queue entry and its persisted `queue.json` record are removed.

## Changes

- **`src/main/download-manager.ts`** — `clearCompleted()` guard extended: completed items with `!merge` (no merge entry — ready-for-merge) are now removable. `'merging'` and crash-recovered `'pending'` still stay so the user can resume them via **Merge finished**.
- **`src/renderer/src/components/views/DownloadsView.vue`** — `hasFinished` includes `'ready-for-merge'`; button label "Clear finished" → "Clear done".
- **`docs/ipc.md`** — refreshed the `download:clear-completed` description (the "skips unmerged" qualifier was wrong).
- **`package.json`** — bump to `v4.0.1`.

## Behavior

| Group state | Before | After |
|---|---|---|
| ready-for-merge (never merged) | stuck forever | **cleared** |
| merging | kept | kept |
| pending (crash-recovered) | kept | kept |
| merged | cleared | cleared |
| merge-failed | cleared | cleared |
| cancelled / failed | cleared | cleared |
| downloading / queued / paused | kept | kept |

## Tests

`test/services/download-manager-clear-completed.test.ts` — 8 cases. Two of them (`removes ready-for-merge groups…` and `clears ready-for-merge alongside merged in a single sweep`) fail on the old `merge?.status === 'completed' || merge?.status === 'failed'` guard and pass on the new `!merge || …` guard. The rest pin the existing behaviors (merged / merge-failed / cancelled / failed clearing; downloading / paused / queued kept; mid-merge and pending preserved).

## Test plan

- [x] `npm run typecheck`
- [x] `npm run lint` (no new warnings)
- [x] `npm run format:check`
- [x] `npm run test` (450 passed)
- [x] `npm run build`
- [ ] Manual: download an episode with auto-merge off → confirm "Clear done" appears and removes the group while leaving the `.mp4` on disk.
- [ ] Manual: delete a `.mp4` after download → confirm the stuck "ready for merge" entry is clearable.
- [ ] Manual regression: fully merged group still clears; failed download still clears.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/renderer/src/components/views/DownloadsView.vue [hunk 1]
```diff
@@ -40,7 +40,7 @@ const hasFailed = computed(() =>
 const hasFinished = computed(() =>
   groups.value.some((g) => {
     const s = groupStatus(g);
-    return s === 'merged' || s === 'failed';
+    return s === 'merged' || s === 'failed' || s === 'ready-for-merge';
```
Comment (mikkerlo) [inline src/renderer/src/components/views/DownloadsView.vue]:
```suggestion
    return s === 'merged' || s === 'failed' || s === 'merge-failed' || s === 'ready-for-merge' || s === 'cancelled';
```
`merge-failed` and `cancelled` should be included here since the backend `clearCompleted` logic now removes them. Conversely, `pending-merge` (if distinguished in `groupStatus`) should be excluded to match the backend's behavior of keeping crash-recovered items.

**Review by mikkerlo — COMMENTED** [review]
The PR successfully implements the clearing of 'ready-for-merge' items with solid backend logic and tests. However, some refinements are needed in the renderer to ensure consistency with the backend clearing logic and to handle edge cases like failed merges and cancelled items.

Specific suggestions:
1. **Update `groupStatus`**: Explicitly handle `pending` and `cancelled` item statuses. Currently, `pending` merges fall through to `ready-for-merge`, which makes the UI offer to clear them even though the backend skips them.
2. **Update `hasFinished`**: Include `merge-failed` and `cancelled` statuses in the computed property so they can also be cleared via the UI.

File: src/renderer/src/components/views/DownloadsView.vue (hunk 1 above)
Comment (akvasha) [reply src/renderer/src/components/views/DownloadsView.vue]: Pushed a fix as a separate commit on top:

- **`hasFinished`** now includes `'merge-failed'` (was missing — well caught).
- **`'cancelled'`**: kept out intentionally. `groupStatus()` never returns `'cancelled'` because cancelled items are filtered out at the top of `getEpisodeGroups()` in the main process (`if (item.status === 'cancelled') continue`), so a cancelled group is invisible in the UI. Including it in `hasFinished` would be dead code.
- **`pending-merge` distinction**: the backend now exposes a new `hasMergeEntry` flag on `EpisodeGroup`. `groupStatus()` returns the new `'pending-merge'` status (label: 'merge interrupted') when a real entry exists with status `'pending'` (i.e. crash-recovered mid-merge), so the UI no longer offers to clear those — matches the backend keeping them. Truly never-merged groups still resolve to `'ready-for-merge'` and are clearable.

Covered by an additional unit test that asserts `hasMergeEntry: false` vs `true` for the two distinct `'pending'` cases.

**Review by mikkerlo — APPROVED** [review]
LGTM after manual testing
